arbitrator/src/beam_search_strategy.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "beam_search_strategy.hpp"
+
+namespace arbitrator
+{
+    std::vector<std::pair<cav_msgs::ManeuverPlan, double>> BeamSearchStrategy::prioritize_plans(std::vector<std::pair<cav_msgs::ManeuverPlan, double>> plans) const
+    {
+        std::sort(plans.begin(), 
+            plans.end(), 
+            [] (std::pair<cav_msgs::ManeuverPlan, double> a, std::pair<cav_msgs::ManeuverPlan, double> b) 
+            {
+                return a.second < b.second;
+            }
+        );
+
+        if (plans.size() > beam_width_)
+        {
+            plans.resize(beam_width_);
```

Comment:
I don't think it should, since I'm only resizing if we're above the specified size. It should just truncate the list, destroying anything in positions `beam_width_` to `plans.size() - 1`. The unit tests seem to confirm this behavior as well.

Reply:
Ah. Sorry, I missed your check above. You are right,